db/post_migrate/20240307180905_migrate_devise_two_factor_secrets.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class MigrateDeviseTwoFactorSecrets < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  class MigrationUser < ApplicationRecord
+    self.table_name = :users
+
+    devise :two_factor_authenticatable,
+           otp_secret_encryption_key: Rails.configuration.x.otp_secret
+
+    include LegacyOtpSecret # Must be after the above `devise` line in order to override the legacy method
```

Comment:
One concern here -- we are going to delete that module at some point in future release.

Maybe that's fine and we just also update this migration to do nothing at that point, or inline it here?